unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -1609,165 +1612,209 @@ private Vector3[] GenerateArcPoints(Vector3 startingPoint, Vector3 origin, float
+        //     TeleportFull(
+        //         position: new Vector3(x, y, z),
+        //         rotation: new Vector3(0, rotation, 0),
+        //         horizon: horizon,
+        //         standing: standing,
+        //         forceAction: forceAction
+        //     );
+        // }
+
+        [ObsoleteAttribute(message: "This action is deprecated. Call TeleportFull(position, ...) instead.", error: false)] 
+        public void TeleportFull(
+            float x, float y, float z,
             Vector3 rotation,
             float horizon,
             bool standing,
             bool forceAction = false
         ) {
-            targetTeleport = new Vector3(x, y, z);
-
-            if (forceAction) {
-                DefaultAgentHand();
-                transform.position = targetTeleport;
-                transform.rotation = Quaternion.Euler(new Vector3(0.0f, rotation.y, 0.0f));
-                if (standing) {
-                    m_Camera.transform.localPosition = standingLocalCameraPosition;
-                } else {
-                    m_Camera.transform.localPosition = crouchingLocalCameraPosition;
-                }
-                m_Camera.transform.localEulerAngles = new Vector3(horizon, 0.0f, 0.0f);
-
-                actionFinished(true);
-                return;
+            TeleportFull(
+                position: new Vector3(x, y, z),
+                rotation: rotation,
+                horizon: horizon,
+                standing: standing,
+                forceAction: forceAction
+            );
+        }
 
-            } else {
-                if (!agentManager.SceneBounds.Contains(targetTeleport)) {
-                    errorMessage = "Teleport target out of scene bounds.";
-                    actionFinished(false);
-                    return;
-                }
+        // keep undocumented until float: rotation is added to Stochastic
+        // public void TeleportFull(
+        //     Vector3 position,
+        //     float rotation,
+        //     float horizon,
+        //     bool standing,
+        //     bool forceAction = false
+        // ) {
+        //     TeleportFull(
+        //         position: position,
+        //         rotation: new Vector3(0, rotation, 0),
+        //         horizon: horizon,
+        //         standing: standing,
+        //         forceAction: forceAction
+        //     );
+        // }
 
-                if (!isPositionOnGrid(targetTeleport)) {
-                    errorMessage = $"Target teleport position {targetTeleport.ToString("F6")} is not" +
-                        $" on the grid of size {gridSize}.";
-                    actionFinished(false);
-                    return;
-                }
+        // has to consider both the arm and standing
+        public void TeleportFull(
+            Vector3 position,
+            Vector3 rotation,
+            float horizon,
+            bool standing,
+            bool forceAction = false
+        ) {
```

Comment:
If we're going by the mentality that a call to `TeleportFull` must always specify all agent parameters, then we need to explicitly override all `base.TeleportFull` methods that do not have a `standing` attribute and have them all throw an error.

Reply:
I have moved TeleportFull and Teleport to protected methods from the base, so this is now no longer necessary.